F821 UndefinedName false positive on annotated global

As of 26d3ff5a3a235cda4883f62e4cbce4096c5c9388 (#1166), Ruff gives an `F821` error here that Flake8 does not. Deleting the `n: int` annotation makes the error disappear.

```python
n: int


def f():
    print(n)


def g():
    global n
    n = 1


g()
f()
```

```console
$ flake8 test.py
$ ruff test.py
Found 1 error(s).
test.py:5:11: F821 Undefined name `n`
```

--- a/resources/test/fixtures/pyflakes/F821_6.py
+++ b/resources/test/fixtures/pyflakes/F821_6.py
@@ -1,0 +1,17 @@
+"""Test: annotated global."""
+
+
+n: int
+
+
+def f():
+    print(n)
+
+
+def g():
+    global n
+    n = 1
+
+
+g()
+f()

--- a/src/check_ast.rs
+++ b/src/check_ast.rs
@@ -1155,7 +1155,13 @@ where
                 // If any global bindings don't already exist in the global scope, add it.
                 let globals = operations::extract_globals(body);
                 for (name, stmt) in operations::extract_globals(body) {
-                    if !self.scopes[GLOBAL_SCOPE_INDEX].values.contains_key(name) {
+                    if self.scopes[GLOBAL_SCOPE_INDEX]
+                        .values
+                        .get(name)
+                        .map_or(true, |index| {
+                            matches!(self.bindings[*index].kind, BindingKind::Annotation)
+                        })
+                    {
                         let index = self.bindings.len();
                         self.bindings.push(Binding {
                             kind: BindingKind::Assignment,
@@ -1207,7 +1213,13 @@ where
                 // If any global bindings don't already exist in the global scope, add it.
                 let globals = operations::extract_globals(body);
                 for (name, stmt) in &globals {
-                    if !self.scopes[GLOBAL_SCOPE_INDEX].values.contains_key(name) {
+                    if self.scopes[GLOBAL_SCOPE_INDEX]
+                        .values
+                        .get(name)
+                        .map_or(true, |index| {
+                            matches!(self.bindings[*index].kind, BindingKind::Annotation)
+                        })
+                    {
                         let index = self.bindings.len();
                         self.bindings.push(Binding {
                             kind: BindingKind::Assignment,
@@ -2717,6 +2729,7 @@ impl<'a> Checker<'a> {
             let mut first_iter = true;
             let mut in_generator = false;
             let mut import_starred = false;
+
             for scope_index in self.scope_stack.iter().rev() {
                 let scope = &self.scopes[*scope_index];
 

--- a/src/pyflakes/mod.rs
+++ b/src/pyflakes/mod.rs
@@ -93,6 +93,7 @@ mod tests {
     #[test_case(CheckCode::F821, Path::new("F821_3.py"); "F821_3")]
     #[test_case(CheckCode::F821, Path::new("F821_4.py"); "F821_4")]
     #[test_case(CheckCode::F821, Path::new("F821_5.py"); "F821_5")]
+    #[test_case(CheckCode::F821, Path::new("F821_6.py"); "F821_6")]
     #[test_case(CheckCode::F822, Path::new("F822.py"); "F822")]
     #[test_case(CheckCode::F823, Path::new("F823.py"); "F823")]
     #[test_case(CheckCode::F831, Path::new("F831.py"); "F831")]

